azure/services/privateendpoints/privateendpoints_test.go

```diff
@@ -0,0 +1,261 @@
+		expect        func(s *mock_privateendpoints.MockPrivateEndpointScopeMockRecorder, r *mock_async.MockReconcilerMockRecorder)
+		expectedError string
+	}{
+		{
+			name:          "create a private endpoint with automatic approval",
+			expectedError: "",
+			expect: func(p *mock_privateendpoints.MockPrivateEndpointScopeMockRecorder, r *mock_async.MockReconcilerMockRecorder) {
+				p.PrivateEndpointSpecs().Return(fakePrivateEndpointSpecs[:1])
+				r.DeleteResource(gomockinternal.AContext(), &fakePrivateEndpoint1, ServiceName).Return(nil)
+				p.UpdateDeleteStatus(infrav1.PrivateEndpointReadyCondition, ServiceName, nil)
+			},
+		},
+		{
+			name:          "noop if no private endpoints specs are found",
+			expectedError: "",
+			expect: func(p *mock_privateendpoints.MockPrivateEndpointScopeMockRecorder, r *mock_async.MockReconcilerMockRecorder) {
+				p.PrivateEndpointSpecs().Return([]azure.ResourceSpecGetter{})
+			},
+		},
+		{
+			name:          "delete multiple private endpoints",
+			expectedError: "",
+			expect: func(p *mock_privateendpoints.MockPrivateEndpointScopeMockRecorder, r *mock_async.MockReconcilerMockRecorder) {
+				p.PrivateEndpointSpecs().Return(fakePrivateEndpointSpecs[:2])
+				r.DeleteResource(gomockinternal.AContext(), &fakePrivateEndpoint1, ServiceName).Return(nil)
+				r.DeleteResource(gomockinternal.AContext(), &fakePrivateEndpoint2, ServiceName).Return(nil)
+				p.UpdateDeleteStatus(infrav1.PrivateEndpointReadyCondition, ServiceName, nil)
+			},
+		},
+		{
+			name:          "error in deleting peering",
+			expectedError: "#: Internal Server Error: StatusCode=500",
+			expect: func(p *mock_privateendpoints.MockPrivateEndpointScopeMockRecorder, r *mock_async.MockReconcilerMockRecorder) {
+				p.PrivateEndpointSpecs().Return(fakePrivateEndpointSpecs[:2])
+				r.DeleteResource(gomockinternal.AContext(), &fakePrivateEndpoint1, ServiceName).Return(nil)
+				r.DeleteResource(gomockinternal.AContext(), &fakePrivateEndpoint2, ServiceName).Return(internalError)
+				p.UpdateDeleteStatus(infrav1.PrivateEndpointReadyCondition, ServiceName, internalError)
+			},
+		},
+		{
+			name:          "not done error in deleting is ignored",
+			expectedError: "#: Internal Server Error: StatusCode=500",
+			expect: func(p *mock_privateendpoints.MockPrivateEndpointScopeMockRecorder, r *mock_async.MockReconcilerMockRecorder) {
+				p.PrivateEndpointSpecs().Return(fakePrivateEndpointSpecs[:3])
+				r.DeleteResource(gomockinternal.AContext(), &fakePrivateEndpoint1, ServiceName).Return(nil)
+				r.DeleteResource(gomockinternal.AContext(), &fakePrivateEndpoint2, ServiceName).Return(internalError)
+				r.DeleteResource(gomockinternal.AContext(), &fakePrivateEndpoint3, ServiceName).Return(notDoneError)
+				p.UpdateDeleteStatus(infrav1.PrivateEndpointReadyCondition, ServiceName, internalError)
+			},
+		},
+		{
+			name:          "not done error in deleting remains",
+			expectedError: "operation type  on Azure resource / is not done",
```

Comment:
This won't work as it will fail with
```
--- FAIL: TestDeletePrivateEndpoints (0.00s)
    --- FAIL: TestDeletePrivateEndpoints/not_done_error_in_deleting_remains (0.00s)
        privateendpoints_test.go:252: 
            Expected
                <azure.OperationNotDoneError>: {
                    Future: {Type: "", ResourceGroup: "", ServiceName: "", Name: "", Data: ""},
                }
            to match error
                <string>: operation type  on Azure resource my-rg/fake-service-endpoint3 is not done
```
I guess it is due to how mocks are created. I'm open to suggestions, currently I left it as `expectedError: "operation type  on Azure resource / is not done"`

Reply:
Yeah, `notDoneError` would also need to be updated to match that new expected message. I could see making `notDoneError` a function that returns an error with the right resource group and name based on the object passed in, something like:
```go
func notDoneError(spec azure.ResourceSpecGetter) error {
	return azure.NewOperationNotDoneError(&infrav1.Future{
		ResourceGroup: spec.ResourceGroupName(),
		Name:          spec.ResourceName(),
	})
}
```
Then to set up the mock for the test case it would look like:
```go
r.CreateOrUpdateResource(gomockinternal.AContext(), &fakePrivateEndpoint2, ServiceName).Return(nil, notDoneError(&fakePrivateEndpoint2))				
p.UpdatePutStatus(infrav1.PrivateEndpointsReadyCondition, ServiceName, notDoneError(&fakePrivateEndpoint2))
```
That involves adding a bit more coordination to make sure the right object is being used in those places though, so I could be convinced this isn't worth it.